### PR TITLE
Remove frontend warnings

### PR DESCRIPTION
**Pull request by shem8**
Remove all kind of warning we have in console when the frontend is starting.
If you want to to test it you can use this branch: https://github.com/ObelusFamily/Anythink-Market-8ira853j/pull/2

### CODE REVIEW

File: .framework/react/frontend/src/reducers/profile.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const reducer = (state = {}, action) => {
   switch (action.type) {
     case PROFILE_PAGE_LOADED:
       return {
-        ...action.payload[0].profile,
+        ...action.payload?.[0]?.profile,
```
Comment (esapir) [inline .framework/react/frontend/src/reducers/profile.js:12]: why did it change?

File: .framework/react/frontend/src/reducers/itemList.js [hunk 2]
```diff
@@ -73,8 +73,8 @@ const reducer = (state = {}, action) => {
       return {
         ...state,
         pager: action.pager,
-        items: action.payload[1].items,
-        itemsCount: action.payload[1].itemsCount,
+        items: action.payload?.[1]?.items,
+        itemsCount: action.payload?.[1]?.itemsCount,
```
Comment (esapir) [inline .framework/react/frontend/src/reducers/itemList.js:77]: why did it change?

File: .framework/react/frontend/src/components/Settings.js [hunk 3]
```diff
@@ -173,4 +139,4 @@ class Settings extends React.Component {
   }
 }
 
-export default connect(mapStateToProps, mapDispatchToProps)(Settings);
+export default connect(mapStateToProps, mapDispatchToProps)(Settings);
```
Comment (esapir) [inline .framework/react/frontend/src/components/Settings.js:142]: add new line in the end of the file

**Review by esapir — APPROVED** [review]

File: .framework/react/frontend/src/components/App.js [hunk 4]
```diff
@@ -30,59 +30,53 @@ const mapDispatchToProps = (dispatch) => ({
...
-      // this.context.router.replace(nextProps.redirectTo);
-      store.dispatch(push(nextProps.redirectTo));
-      this.props.onRedirect();
+const App = (props) => {
+  const { redirectTo, onRedirect, onLoad } = props;
+
+  useEffect(() => {
+    if (redirectTo) {
+      store.dispatch(push(redirectTo));
+      onRedirect();
```
[the remark this answers is not in the document]
Comment (shem8) [reply .framework/react/frontend/src/components/App.js:39]: it wasn't before so didn't want to add

**Pull request merged**